hack/update-codegen.sh

```diff
@@ -37,7 +37,7 @@ informergen=$PWD/$3
 listergen=$PWD/$4
 
 ensure-in-gopath() {
-  fake_gopath=$(mktemp -d --tmpdir codegen.gopath.XXXX)
+  fake_gopath=$(mktemp -d -t codegen.gopath.XXXX)
```

Comment:
_technically_ these flags have slightly different meaning FYI. Also -t is deprecated so of course we should use that.

More seriously I don't think we need -t. We're specifying a template so it should just be `mktemp -d $template`

http://man7.org/linux/man-pages/man1/mktemp.1.html

https://www.unix.com/man-page/osx/1/mktemp/

Reply:
Example of this which runs fine on Mac / Linux AFAIK, https://github.com/kubernetes-sigs/kind/blob/b4465e9de7b4e0cc25aab705e5eb6b53dfd39737/hack/verify-generated.sh#L30

Reply:
I added the --tmpdir initlaly because it doesn't work without it. Without the `-t` flag then it winds up creating a path relative to PWD (aka `codegen.gopath.XXX` not `/tmp/codegen.gopath.XXX`). 

The example you give doesn't seem comparable: a) it sets TMPDIR b) the template is an absolute path starting with `git rev-parse --show-toplevel`

Reply:
huh all the docs say it should do relative to $TMPDIR or `/tmp` if $TMPDIR is unset, but testing locally it does `$PWD` if TMPDIR is unset for me as well ...

resolving